### PR TITLE
update documentation to cover PMI

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Add documentation for `jit-diff diff --pmi`, `jit-dasm-pmi`, and `pmi`.

Also a blurb on how to think about the different kinds of diffs now available.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@BruceForstall PTAL
cc @dotnet/jit-contrib

**Comment by CarolEidt** [comment]
This looks great - thanks @AndyAyersMS !

File: README.md [hunk 1]
```diff
@@ -11,9 +11,9 @@ or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any addi
 
 Current tools include:
 
-1. [Assembly diffs](doc/diffs.md): jit-diff, jit-dasm, jit-analyze.
+1. [Assembly diffs](doc/diffs.md): jit-diff, jit-dasm, jit-dasm-pmi, jit-analyze, pmi.
```
Comment (BruceForstall) [inline README.md]: Well, "pmi" on its own doesn't fit in the "Assembly diffs" category. Maybe it needs a:

"JIT testing tools"

category.

File: doc/diffs.md [hunk 2]
```diff
@@ -468,3 +535,34 @@ diffing.  Note: The mscorlib.dll is removed, as this assembly should be updated
 the selected base runtime that is under test, for consistency. To add particular packages 
 to the set you diff, add their dependencies to the project.json in this project and 
 they will be pulled in and published in the standalone directory './fx'.
+
+## pmi
```
Comment (BruceForstall) [inline doc/diffs.md]: I think PMI deserves its own file, since it doesn't need to be tied to assembly diffing.

File: doc/diffs.md [hunk 3]
```diff
@@ -63,26 +65,37 @@ build.sh [-b <BUILD TYPE>] [-f] [-h] [-p] [-t <TARGET>]
...
 * optionally, `COMPlus_NgenGCDump`
 
-Generating "diffs" involves generating assembly code output for both a baseline and a "diff" JIT,
-and comparing the results.
+Generating "diffs" involves generating assembly code output for both a baseline
+and a "diff" JIT, and comparing the results.
+
+Passing the `--pmi` option to `jit-diffs` will instead use reflection to jit each
+method in the assembly, setting these options:
+* `COMPlus_Disasm`
```
Comment (BruceForstall) [inline doc/diffs.md]: These variables are wrong (hopefully the code is right). It should be:
```
COMPLUS_JitDisasm
COMPLUS_JitUnwindDump
COMPLUS_JitEHDump
COMPLUS_JitGCDump
```

Comment (AndyAyersMS) [reply doc/diffs.md]: Just typos in the doc, fixed.

File: doc/diffs.md (hunk 2 above)
Comment (AndyAyersMS) [reply doc/diffs.md]: Moved it to a `Tools` page with the latest commit.

**Comment by AndyAyersMS** [comment]
One more update to fix the `jit-diffs` in app usage to show pmi examples. Sorry about the force push but I wanted to rebase on top of the actual PMI changes.

**Pull request merged**